### PR TITLE
WebApi version

### PR DESCRIPTION
**Pull request by CactusGreen**
Moved database to a WebApi project, it can now be accessed through WebApi. There might be some issues with stability and might have made some unnecessary changes . Would be nice to see if this is fine or should be remade.

### CODE REVIEW

**Comment by CactusGreen** [comment]
I noticed that i accidentally changed namespace in all project. It shouldn't break anything for now but will have to be fixed later.

**Comment by CactusGreen** [comment]
Okay apparently that namespace change breaks a lot of things when trying to merge conflicts. Will have to fix it before merging this to master.

**Comment by joklek** [comment]
Can you write how it can be used?

File: BeerOverflowWindowsApp/WebApiAccess.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            using (var client = new HttpClient())
+            {
+                client.BaseAddress = new Uri("http://localhost:1726/");
+                client.DefaultRequestHeaders.Accept.Clear();
+                client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+                var json = JsonConvert.SerializeObject(bar);
+                var content = new StringContent(json, Encoding.UTF8, "application/json");
+                var response = client.PostAsync("Api/Data/GetBarRatings", content);
+                var ab = response.Result.Content.ReadAsStringAsync();
```
Comment (joklek) [inline BeerOverflowWindowsApp/WebApiAccess.cs]: variable name not clear.

File: WebApi/Database/DatabaseManager.cs [hunk 2]
```diff
@@ -34,8 +34,12 @@ public void SaveBarRating(BarData barToRate, int rating)
             }
         }
 
-        public BarDataModel GetAllBarData(BarDataModel localBars)
+        public List<BarData> GetAllBarData(List<BarData> localBars)
         {
+            //if (localBars == null)
```
Comment (joklek) [inline WebApi/Database/DatabaseManager.cs]: Commented code needed? If not, please remove.

File: BeerOverflowWindowsApp/MainWindow.cs [hunk 3]
```diff
@@ -204,10 +210,11 @@ private string GetRadius()
         private void ManualBarRating_Click(object sender, EventArgs e)
         {
             var rating = manualBarRating.Rating;
-            if (_selectedBar != null && rating != "" && int.TryParse(rating, out int ratingNumber))
+            int ratingNumber;
+            if (_selectedBar != null && rating != "" && int.TryParse(rating, out ratingNumber))
             {
-                Task.Run(() => _barRating.AddRating(_selectedBar, ratingNumber)).Wait();
-                Task.Run(() => _selectedBar.Ratings = new DatabaseManager().GetBarRatings(_selectedBar)).Wait();
```
Comment (joklek) [inline BeerOverflowWindowsApp/MainWindow.cs]: Is it possible to re-add the  "Task.Run(() => ...).Wait()" part for more responsivness?

File: BeerOverflowWindowsApp/MainWindow.cs [hunk 4]
```diff
@@ -121,18 +120,25 @@ private void ClearSelection()
                 }
                 result.RemoveDuplicates();
                 result.RemoveBarsOutsideRadius(radius);
-                await Task.Run(() => result = new DatabaseManager().GetAllBarData(result));
+                //await Task.Run(() => 
+                result = (BarDataModel)WebApiAccess.GetAllBarData(result);
```
Comment (joklek) [inline BeerOverflowWindowsApp/MainWindow.cs]: Same here about the Task.Run(), as elsewhere. Can you re-add it, or does it not work?

**Pull request merged**